src/ProcessStamp.php

```diff
@@ -54,6 +71,129 @@ public static function firstOrCreateByProcess(array $process, ?string $hash = nu
      */
     public static function makeProcessHash(array $process) : string
     {
-        return sha1($process['type'].'-'.$process['name']);
+        return sha1($process['type'] . '-' . trim($process['name']));
+    }
+
+    /**
+     * Get the parent of the command, if there is one.
+     *
+     * @return BelongsTo|null
+     */
+    public function parent() : ?BelongsTo
+    {
+        return $this->belongsTo(self::class, 'parent_id');
+    }
+
+    /**
+     * Get any child processes.
+     *
+     * @return HasMany|null
+     */
+    public function children() : ?HasMany
+    {
+        return $this->hasMany(self::class, 'parent_id');
+    }
+
+    /**
+     * @return string
+     */
+    public static function getType() : string
+    {
+        if (isset($_SERVER['REQUEST_URI'])) {
```

Comment:
Should we use `\Request::server()` here instead?

Reply:
Generally yes, but if you're using scripts outside of a Laravel application (like legacy code) they don't invoke the Laravel request lifecycle and `Request::server()` can return null results

Reply:
Good point, possibly leave that as a comment for future refactors so we don't accidentally update.